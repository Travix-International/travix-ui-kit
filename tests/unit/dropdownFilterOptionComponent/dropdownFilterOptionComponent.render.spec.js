import { shallow } from 'enzyme';
import React from 'react';
import DropdownFilterOptionComponent from '../../../components/dropDown/dropdownFilterOptionComponent';

describe('dropdownFilterOptionComponent', () => {
  describe('#render()', () => {
    it('should render filter option', () => {
      const option = {
        label: 'Three',
        value: '3',
      };

      const onSelect = jest.fn();
      const children = 'Three';

      const wrapper = shallow(
        <DropdownFilterOptionComponent
          onSelect={onSelect}
          option={option}
        >
          {children}
        </DropdownFilterOptionComponent>
      );

      expect(onSelect).not.toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render disabled and checked filter option', () => {
      const option = {
        label: 'Three',
        value: '3',
        disabled: true,
        checked: true,
      };

      const onSelect = jest.fn();
      const children = 'Three';

      const wrapper = shallow(
        <DropdownFilterOptionComponent
          onSelect={onSelect}
          option={option}
        >
          {children}
        </DropdownFilterOptionComponent>
      );

      expect(onSelect).not.toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
