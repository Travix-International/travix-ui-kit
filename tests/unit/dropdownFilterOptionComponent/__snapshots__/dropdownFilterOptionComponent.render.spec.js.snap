// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdownFilterOptionComponent #render() should render disabled and checked filter option 1`] = `
<div
  className="ui-dropdown-filter__option ui-dropdown-filter__option_is-disabled"
>
  <Checkbox
    checked={true}
    disabled={true}
    name="ui-dropdown-filter-3"
    onChange={[Function]}
  >
    Three
  </Checkbox>
</div>
`;

exports[`dropdownFilterOptionComponent #render() should render filter option 1`] = `
<div
  className="ui-dropdown-filter__option"
>
  <Checkbox
    checked={false}
    disabled={false}
    name="ui-dropdown-filter-3"
    onChange={[Function]}
  >
    Three
  </Checkbox>
</div>
`;
